pkg/http/request_test.go

```diff
@@ -0,0 +1,40 @@
+package http
+
+import (
+	"io"
+	h "net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_requestReadMultiple(t *testing.T) {
+	var req = require.New(t)
+	r, _ := h.NewRequest("POST", "/foo", strings.NewReader(`foo body`))
+
+	rs, err := NewBufferedReader(r.Body)
+
+	req.NoError(err)
+	req.Equal(`foo body`, must(io.ReadAll(rs)))
+	req.Equal(`foo body`, must(io.ReadAll(rs)))
+}
```

Comment:
Can we have a test case with a larger payload?
I'm looking at `io.ReadAll` implementation and it reads in `512 byte` chunks; so if your payload exceeds that (and I do not miss interpret the implementation) it should start breaking.

Reply:
Nice catch, will check how to implement Seeker, it does look like the Read limitation will cause trouble.